### ISSUE TITLE
Crashing while adding widgets

### ISSUE THREAD

**Issue by El-Castellano** [Deletescape-Media/Lawnchair#2277]
- Yesterday I was trying to add "another widget", a customizable at a glance widget but the launcher kept crashig and reseting to the main One UI 3.1 launcher
Phone: Samsung Galaxy a51 a515f, running android 11 & One UI 3.1

**Comment by dlphesigns** [comment]
Can confirm the bug above is affecting me too. 
Phone: Samsung Galaxy A52s (model: SM-A528B/DS)
Platform: Samsung-skinned Android 11, skin version One UI 3.1

**Comment by ueen** [comment]
Pixel 4a 5G same, also touching on widget crashes

**Comment by suphon-t** [comment]
Can you guys on A11 send a crash log here?

**Comment by dlphesigns** [comment]
Wait, reinstalled and it works for me now.

**Comment by El-Castellano** [comment]
> Wait, reinstalled and it works for me now.

I did so. But now some other widgets make it happen (Google tasks, Samsung weather, KWGT etc.)

**Comment by El-Castellano** [comment]
The crash only happens when the lawnchair is the main launcher.
Found out rn.

**Comment by dlphesigns** [comment]
> The crash only happens when the lawnchair is the main launcher. Found out rn.

I had realised this behaviour too days ago, and I apologise for not reporting the full variant here.
The application will only force quit upon widget placement if in Settings 'None' or 'Lawnchair' are specified as home apps. It will not force quit upon placement if the system home app or any other launcher is specified as the home function.
Thank you for reporting on this. Nice day!

**Comment by dlphesigns** [comment]
> Can you guys on A11 send a crash log here?

I'm afraid I don't know how to obtain one. Mind to share, please?

**Comment by ueen** [comment]
Alpha 3

```
bug report 02.11.2021 14:55:30
version: 12.0 Alpha 3 (120003)
commit: 7e4d7bb
build.brand: google
build.device: bramble
build.display: SP1A.211105.003
build.fingerprint: google/bramble/bramble:12/SP1A.211105.003/7757856:user/release-keys
build.hardware: bramble
build.id: SP1A.211105.003
build.manufacturer: Google
build.model: Pixel 4a (5G)
build.product: bramble
build.type: user
version.codename: REL
version.incremental: 7757856
version.release: 12
version.sdk_int: 31
display.density_dpi: 420

error: Uncaught exception

java.lang.RuntimeException: item: LauncherAppWidgetInfo(id=2 type=WIDGET container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@1a1bf6a
workspace {
  grid_x: 0
  grid_y: 0
  page_index: 0
} targetComponent=ComponentInfo{com.duskjockeys.climateweatherwidget/com.duskjockeys.climateanimatedweatherwidget.CliMateAnimatedWeatherWidget} screen=0 cell(0,0) span(4,3) minSpan(4,3) rank=0 user=UserHandle{0} title=null providerName=ComponentInfo{com.duskjockeys.climateweatherwidget/com.duskjockeys.climateanimatedweatherwidget.CliMateAnimatedWeatherWidget} appWidgetId=22)modelItem: WorkspaceItemInfo(id=2 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@1a1ca63
hotseat {
  index: 1
} targetComponent=ComponentInfo{com.textra/com.mplus.lib.ui.main.Main} screen=1 cell(1,0) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Textra)Error: ItemInfo passed to checkItemInfo doesn't match original
	at com.android.launcher3.model.ModelWriter.addItemToDatabase(Unknown Source:30)
	at com.android.launcher3.Launcher.completeAddAppWidget(Unknown Source:83)
	at com.android.launcher3.Launcher$4.run(Unknown Source:9)
	at com.android.launcher3.Workspace$6.run(Unknown Source:12)
	at com.android.launcher3.dragndrop.DragLayer$2.onAnimationEnd(Unknown Source:4)
	at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:555)
	at android.animation.ValueAnimator.endAnimation(ValueAnimator.java:1250)
	at android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1492)
	at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
	at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
	at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1035)
	at android.view.Choreographer.doCallbacks(Choreographer.java:845)
	at android.view.Choreographer.doFrame(Choreographer.java:775)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7838)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by suphon-t** [comment]
Can anyone try and see if it's fixed in [this](https://github.com/LawnchairLauncher/lawnchair/actions/runs/1410970221) build?

**Comment by konradstrg** [comment]
can confirm this affects me too (Samsung Galaxy A51, Android 11)

**Comment by MobleVortex** [comment]
Can't refresh widgets such as simple crypto widget or robinhood. Running LineageOS 19.1 May 21st build.

```
Lawnchair bug report May 26, 2022 8:49:05 PM
version: 12.1.0 Alpha 3 (12010003)
commit: e7f0269
build.brand: google
build.device: bonito
build.display: lineage_bonito-userdebug 12 SP2A.220505.002 8f0c2ac331
build.fingerprint: google/bonito/bonito:12/SP2A.220505.002/8353555:user/release-keys
build.hardware: bonito
build.id: SP2A.220505.002
build.manufacturer: Google
build.model: Pixel 3a XL
build.product: =bonito
build.type: user
version.codename: REL
version.incremental: 8f0c2ac331
version.release: 12
version.sdk_int: 32
display.density_dpi: 400
isRecentsEnabled: true

error: Uncaught exception

java.lang.SecurityException: Permission Denial: registerRemoteAnimationForNextActivityStart from pid=2221, uid=10231 requires android.permission.CONTROL_REMOTE_APP_TRANSITION_ANIMATIONS
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.os.Parcel.readException(Parcel.java:2334)
	at android.app.IActivityTaskManager$Stub$Proxy.registerRemoteAnimationForNextActivityStart(IActivityTaskManager.java:4403)
	at com.android.launcher3.uioverrides.QuickstepInteractionHandler.onInteraction(QuickstepInteractionHandler.java:12)
	at android.appwidget.AppWidgetHostView.lambda$getHandler$1$AppWidgetHostView(AppWidgetHostView.java:882)
	at android.appwidget.AppWidgetHostView$$ExternalSyntheticLambda2.onInteraction(Unknown Source:4)
	at android.widget.RemoteViews$RemoteResponse.handleViewInteraction(RemoteViews.java:6555)
	at android.widget.RemoteViews$RemoteResponse.access$200(RemoteViews.java:6403)
	at android.widget.RemoteViews$SetOnClickResponse.lambda$apply$0$RemoteViews$SetOnClickResponse(RemoteViews.java:1199)
	at android.widget.RemoteViews$SetOnClickResponse$$ExternalSyntheticLambda0.onClick(Unknown Source:4)
	at android.view.View.performClick(View.java:7455)
	at android.view.View.performClickInternal(View.java:7432)
	at android.view.View.access$3700(View.java:835)
	at android.view.View$PerformClick.run(View.java:28810)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7870)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActivityManagerService.enforceCallingPermission(ActivityManagerService.java:5481)
	at com.android.server.am.ActivityManagerService$LocalService.enforceCallingPermission(ActivityManagerService.java:15637)
	at com.android.server.wm.ActivityTaskManagerService.registerRemoteAnimationForNextActivityStart(ActivityTaskManagerService.java:3631)
	at android.app.IActivityTaskManager$Stub.onTransact(IActivityTaskManager.java:2076)
	at com.android.server.wm.ActivityTaskManagerService.onTransact(ActivityTaskManagerService.java:5128)

```

**Comment by SuperDragonXD** [comment]
No logs in this issue, and should be fixed in Lawnchair 12.1 Alpha 4. Closing this issue.
